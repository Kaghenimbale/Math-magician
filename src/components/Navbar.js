import { NavLink } from 'react-router-dom';
import styles from '../styles/Navbar.module.css';

const Navbar = () => {
  const links = [
    {
      path: '/',
      element: 'Home',
    },
    {
      path: 'calculator',
      element: 'Calculator',
    },
    {
      path: 'quotes',
      element: 'Quotes',
    },
  ];
  return (
    <nav className={styles.navbar}>
      <h2>Math Magicians</h2>
      <ul className={styles.navlist}>
        {links.map((link) => (
          <li key={link.element}>
            <NavLink className={styles.navlink} to={link.path}>
              {link.element}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
