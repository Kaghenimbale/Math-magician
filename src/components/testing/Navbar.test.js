import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Navbar', () => {
  test('Render Navbar component', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should redirect and update dom', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(screen.getByText(/Home/i)).toBeInTheDocument();
  });
});
