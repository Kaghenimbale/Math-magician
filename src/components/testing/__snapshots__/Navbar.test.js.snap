// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Render Navbar component 1`] = `
<nav
  className="navbar"
>
  <h2>
    Math Magicians
  </h2>
  <ul
    className="navlist"
  >
    <li>
      <a
        aria-current="page"
        className="navlink active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="navlink"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className="navlink"
        href="/quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;
